controllers/topology/desired_state.go

```diff
@@ -58,7 +60,17 @@ func (r *ClusterReconciler) computeDesiredState(_ context.Context, class *cluste
 	// InfrastructureCluster and the ControlPlane objects generated by the previous step.
 	desiredState.cluster = computeCluster(current, desiredState.infrastructureCluster, desiredState.controlPlane.object)
 
-	// TODO: implement generate desired state for machine deployments
+	// Compute the desired state of the MachineDeployment objects for the worker nodes.
+	if len(current.cluster.Spec.Topology.Workers.MachineDeployments) > 0 {
+		desiredState.machineDeployments = map[string]machineDeploymentTopologyState{}
```

Comment:
Should we initialize this outside to avoid nil maps?

Reply:
I think the intended use is to loop over the `machineDeployments` map from the desired state (go over them one at a time, if any), which will end gracefully if the map is nil.
So IMO either way this should work.